witness/src/Dscp/Witness/Mempool/Logic.hs

```diff
@@ -0,0 +1,128 @@
+-- | Memory pool. TBD.
+
+module Dscp.Witness.Mempool.Logic
+    ( Mempool(..)
+    , MempoolVar
+    , newMempoolVar
+    , addTxToMempool
+    , takeTxsMempool
+    , isInMempool
+    ) where
+
+import Dscp.Core.Foundation (GTxWitnessed)
+import qualified Dscp.Snowdrop as SD
+import qualified Dscp.Snowdrop.Configuration as Conf
+import qualified Dscp.Snowdrop.Storage.Avlp as AVLP
+import Dscp.Witness.Mempool.Type
+import qualified Dscp.Witness.SDLock as Lock
+import Loot.Base.HasLens (HasLens', lensOf)
+import qualified Snowdrop.Core as SD
+import qualified Snowdrop.Model.Execution as SD (dmaAccessActions)
+import qualified Snowdrop.Model.Mempool as Pool
+import qualified Snowdrop.Util as SD
+import qualified UnliftIO
```

Comment:
Name at least one reason why `UnliftIO.MonadUnliftIO` is better than `MonadUnliftIO`.